doc/design/initialization.md

```diff
@@ -0,0 +1,48 @@
+# Initialization of Parameters
+
+## Motivation
+
+PaddlePaddle needs a way to init parameters.
+
+## Challenge
+
+Initialization operators must run once and only one; otherwise, every iteration would clear the parameters.
+
+## Proposals
+
+### Proposal 1: Two seperate `ProgramDesc`.
```

Comment:
this seems simple and workable.

The resource which survived across multiple block/program is the `Variables` in `Scopes`.

A program is the container of multiple Blocks, more than one program seems not a problem.